evm/src/executor/fork/cache.rs

```diff
@@ -0,0 +1,234 @@
+//! Cache related abstraction
+use ethers::types::{Address, H256, U256};
+use parking_lot::RwLock;
+use revm::AccountInfo;
+use serde::{ser::SerializeMap, Deserialize, Deserializer, Serialize, Serializer};
+use std::{collections::BTreeMap, fs, io::BufWriter, path::PathBuf, sync::Arc};
+use tracing::{trace, trace_span, warn};
+use tracing_error::InstrumentResult;
+
+pub type StorageInfo = BTreeMap<U256, U256>;
+
+/// A shareable Block database
+#[derive(Clone)]
+pub struct BlockchainDb {
+    /// Contains all the data
+    db: Arc<MemDb>,
+    /// metadata of the current config
+    meta: Arc<RwLock<BlockchainDbMeta>>,
+    /// the cache that can be flushed
+    cache: Arc<dyn BlockCacheDB>,
```

Comment:
this feels  a bit weird because it's almost like a circular dependency;

BlockchainDb -> JsonBlockCache -> the meta and db fields are repeated.

Could we remove the cache trait and make the cache here the pathbuf? We can re evaluate a refactor in the future imo